### ISSUE TITLE
The Result of Graph Convolutions For Tox21

### ISSUE THREAD

**Issue by lch85** [deepchem/deepchem#1627]
Dear all,
I'm new to deepchem and graph convolution.
I tried to run the source of "Graph Convolutions For Tox21" following URL.
https://deepchem.io/docs/notebooks/graph_convolutional_networks_for_tox21.html

Why the result changes every time? Can any one help me?
How do I fix the results, when using the same training data and test data.

The source is hear.
```
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals
import numpy as np
import tensorflow as tf
import deepchem as dc
from deepchem.models.tensorgraph.models.graph_models import GraphConvModel

# Load Tox21 dataset
tox21_tasks, tox21_datasets, transformers = dc.molnet.load_tox21(featurizer='GraphConv')
train_dataset, valid_dataset, test_dataset = tox21_datasets

model = GraphConvModel(
    len(tox21_tasks), batch_size=50, mode='classification')
# Set nb_epoch=10 for better results.
model.fit(train_dataset, nb_epoch=1)

metric = dc.metrics.Metric(
    dc.metrics.roc_auc_score, np.mean, mode="classification")
print("Evaluating model")
train_scores = model.evaluate(train_dataset, [metric], transformers)
print("Training ROC-AUC Score: %f" % train_scores["mean-roc_auc_score"])
valid_scores = model.evaluate(valid_dataset, [metric], transformers)
print("Validation ROC-AUC Score: %f" % valid_scores["mean-roc_auc_score"])
```

And the results are hear.
```
---------------------------------------------------------------------------
Evaluating model
computed_metrics: [0.7834564076711283, 0.8445563383108542, 0.8184271866527277, 0.7900669478617008, 0.695882232343717, 0.7691039663418828, 0.7520419590962076, 0.6847741708719339, 0.762719841595735, 0.654384747497528, 0.8007863812262577, 0.7516379985456174]
Training ROC-AUC Score: 0.758987
computed_metrics: [0.6921288279819917, 0.746114417989418, 0.8022573157966595, 0.7186705643111337, 0.6032500000000001, 0.6489261269766344, 0.5858428521979924, 0.6898940792409403, 0.7400988933364727, 0.5624101042790363, 0.7562576349134111, 0.6918604651162792]
Validation ROC-AUC Score: 0.686476
--------------------------------------------------------------------------
Evaluating model
computed_metrics: [0.8171508974408691, 0.8491101176140012, 0.8325430771933104, 0.7727595017993552, 0.6904507020401831, 0.7568917055767578, 0.7638121639920638, 0.686507494162838, 0.7664289616725708, 0.6875662266016689, 0.7925170742845369, 0.7387257369804777]
Training ROC-AUC Score: 0.762872
computed_metrics: [0.7353683273141047, 0.8115906084656086, 0.8104946980385017, 0.7526054905948145, 0.6354318181818182, 0.6368893084729762, 0.597005884389062, 0.685314356842575, 0.7029903461266777, 0.5718941028407047, 0.7429639033270257, 0.655770887166236]
Validation ROC-AUC Score: 0.694860
--------------------------------------------------------------------------
```

**Comment by peastman** [comment]
There are two parts that aren't deterministic.  First, all the model parameters get randomly initialized, so that will be different each time you run it.  Second, `fit()` runs through the dataset in a random order, so that also will be different each time.

I think if you call `tf.set_random_seed()` and `np.random.seed()` at the start of the run to set both of them to fixed values, then results just be reproducible.

**Comment by lch85** [comment]
Dear peastman,

Thanks for the response.
I modified the code like below, then results could be reproducible.
Thank you very much. I'm going to close this question now.

```
np.random.seed(0)
model = GraphConvModel(len(tox21_tasks), batch_size=50, mode="classification", random_seed=0)
model.fit(train_dataset, nb_epoch=1, deterministic=True)
```
